tests/integration_tests/performance/test_cpu_template_benchmark.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""A benchmark that checks for regression of CPU template operations."""
+
+import json
+import logging
+import os
+import platform
+import shutil
+from pathlib import Path
+
+import pytest
+
+from framework import utils
+from framework.defs import FC_WORKSPACE_DIR
+from host_tools import proc
+
+BENCHMARK_DIRECTORY = "{}/src/vmm".format(FC_WORKSPACE_DIR)
+DEFAULT_BUILD_TARGET = "{}-unknown-linux-musl".format(platform.machine())
+
+PROC_MODEL = proc.proc_type()
+
+NSEC_IN_MSEC = 1000000
+
+BASELINES = {
+    "Intel": {
+        "deserialize": {"target": 0.025, "delta": 0.02}  # milliseconds
+    },
+    "AMD": {
+        "deserialize": {"target": 0.025, "delta": 0.02}  # milliseconds
+    },
+    "ARM": {
+        "deserialize": {"target": 0.0015, "delta": 0.001}  # milliseconds
+    },
+}
+
+
+def _check_statistics(directory, mean):
+    proc_model = [item for item in BASELINES if item in PROC_MODEL]
+    assert len(proc_model) == 1, "Could not get processor model!"
+
+    measure = BASELINES[proc_model[0]]["deserialize"]
+    target, delta = measure["target"], measure["delta"]
+    assert target == pytest.approx(
+        mean, abs=delta, rel=1e-6
```

Comment:
why do we need both `abs` and `rel` here? Is `abs` not going to be always greater than `1e-6`?

Reply:
Not necessarily. It depends on the data set size. I would like to see these tests expanded to check small versus large data sets. On very small data sets the relative margin is very small.

Reply:
Hm. I'm not sure I understand the influence of the dataset size. At this point the samples are already reduced to a scalar value (mean). We are just comparing 2 numbers here. How would the dataset size affect the significance of the `rel` value without being "noticed" by the `abs`?

Reply:
Much larger data sets would have larger bands of variance, well beyond the delta defined here. This is a bit of a short-cut that `test_versioned_serialization_benchmark.py` takes as well, so that we don't have to configure deltas for every dataset size. I included the parameter here as it makes it convenient to add changing datasets. I can remove it if you prefer, and the relativity issue can be address when extra data sets are introduced?

Reply:
I think it is fine is as it is, but could we document our reasoning in a code comment to aid decisions in the future?